modules/platforms/cpp/pom.xml

```diff
@@ -0,0 +1,115 @@
+                <plugins>
+                    <plugin>
+                        <groupId>com.googlecode.cmake-maven-project</groupId>
+                        <artifactId>cmake-maven-plugin</artifactId>
+                        <configuration>
+                            <downloadBinaries>false</downloadBinaries>
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <id>cmake-generate</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>generate</goal>
+                                </goals>
+                                <configuration>
+                                    <sourcePath>${project.basedir}</sourcePath>
+                                    <targetPath>${project.build.directory}/cpp</targetPath>
+                                    <generator>Unix Makefiles</generator>
+                                    <options>
+                                        <option>-DCMAKE_BUILD_TYPE=Release</option>
+                                    </options>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>cmake-build</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>compile</goal>
+                                </goals>
+                                <configuration>
+                                    <projectDirectory>${project.build.directory}/cpp</projectDirectory>
+                                    <options>
+                                        <option>-j</option>
+                                    </options>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-native-library</id>
+                        <phase>prepare-package</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/classes/</outputDirectory>
+                            <resources>
+                                <resource>
+                                    <directory>${project.build.directory}/cpp/</directory>
+                                    <includes>
+                                        <include>**/*.so</include>
```

Comment:
Two sets of questions here as well:
- Do we only support `.so` binaries? Why not include `.dll`? Has this been discussed?
- Usually binaries go to specific folders for specific architectures. Is this the case here? Has the list of supported architectures been discussed?

Reply:
Support of other platforms will be done later. Let's start with something. Ignite-3 still has time to mature.

Reply:
This will be addressed under IGNITE-17401 ticket.